### DISCUSSION TITLE
Microgame model

### DISCUSSION THREAD

**Discussion by paulloz** [NEW-CYLANDIA/little-warioware#5] [Technical design]
I am starting to think about how we're going to model the microgames.  

Here's the basic idea. Microgames are defined by two things:
* A `MicrogameDefinition` object derived from [Resource](https://docs.godotengine.org/en/stable/classes/class_resource.html)
* A `Microgame` object derived from [Node](https://docs.godotengine.org/en/stable/classes/class_node.html)

The former contains basic information about the microgame: its name, its author's name, etc. and a reference to the scene object.  
The latter contains the actual gameplay for the microgame.

![mermaid-diagram-2023-02-22-103235](https://user-images.githubusercontent.com/437025/220579827-dae145ac-3cc1-4509-bd29-774784e319ee.svg)

**Comment by iznaut** [comment]
i think this is good! there's a lot of metadata that we may or may not want to capture - e.g. author would be an array for collabs, maybe also displaying roles (artist, designer, programmer, etc) but i assume that's the value of keeping that part separated like this (so we can add/modify meta fields without messing with gameplay content)?
